RIOT-OS/RIOT
# dist/tools: add DOSE Linux tool

### Contribution description

This adds an `ethos` derived tool to monitor a DOSE network by exposing it via a TAP interface. The main use case it to see the traffic on the DOSE bus in Wireshark, but preliminary write support was added too.

This does not include collision detection, so use it with caution.

### Testing procedure

Connect a USB-UART adapter to a CAN transceiver and hook the transceiver up to your DOSE network, then run

    sudo ./dist/tools/dose/start_network.sh --write /dev/ttyUSB1 dose0 1000000

will create a new `dose0` interface that makes use of the UART at `/dev/ttyUSB1` with a 1 MHz bitrate and write support enabled.

If you only want to monitor a network and not interfere with it, leave the write support disabled.

You should now be able to ping DOSE nodes from Linux:

```
% ping -c3 ff02::1%dose0
PING ff02::1%dose0(ff02::1%dose0) 56 data bytes
64 bytes from fe80::18cf:efff:fe0d:9a97%dose0: icmp_seq=1 ttl=64 time=0.055 ms
64 bytes from fe80::18cf:efff:fe0d:9a97%dose0: icmp_seq=1 ttl=64 time=20.9 ms
64 bytes from fe80::acde:b5ff:fe69:1703%dose0: icmp_seq=1 ttl=64 time=20.9 ms
64 bytes from fe80::64f6:d6ff:fe18:129a%dose0: icmp_seq=1 ttl=64 time=20.9 ms
64 bytes from fe80::18cf:efff:fe0d:9a97%dose0: icmp_seq=2 ttl=64 time=0.074 ms
64 bytes from fe80::18cf:efff:fe0d:9a97%dose0: icmp_seq=2 ttl=64 time=21.5 ms
64 bytes from fe80::acde:b5ff:fe69:1703%dose0: icmp_seq=2 ttl=64 time=21.5 ms
64 bytes from fe80::64f6:d6ff:fe18:129a%dose0: icmp_seq=2 ttl=64 time=21.5 ms
64 bytes from fe80::18cf:efff:fe0d:9a97%dose0: icmp_seq=3 ttl=64 time=0.074 ms

--- ff02::1%dose0 ping statistics ---
3 packets transmitted, 3 received, +6 duplicates, 0% packet loss, time 2003ms
rtt min/avg/max/mdev = 0.055/14.179/21.547/9.981 ms
```

### Issues/PRs references
